lib/Zonemaster/LDNS/IDN.pm

```diff
@@ -0,0 +1,47 @@
+package Zonemaster::LDNS::IDN;
+
+use strict;
+use warnings;
+
+use Carp;
+
+use parent 'Exporter';
+our @EXPORT_OK = qw[has_idn to_idn];
+
+my $idn_available = 0;
+
+eval {
+    use Net::LibIDN2 ':all';
+};
+
+if ( ! $@ ) {
+    $idn_available = 1;
+}
```

Comment:
Have you considered wrapping the `eval + if` in a BEGIN block and replacing `use` with `require`? I believe that would reduce the amount of magic going on. (See docs for [use](https://perldoc.perl.org/functions/use) and [BEGIN](https://perldoc.perl.org/perlmod#BEGIN%2C-UNITCHECK%2C-CHECK%2C-INIT-and-END).)